### ISSUE TITLE
[HOLD for payment 2024-04-09] [$500] [Remove Free Workspace Creation] Update Workspace Creation in NewDot to Create Collect Workspaces

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#37153]
- Held on https://github.com/Expensify/App/issues/37149 so that the new notifications are in place on the server
- Held on https://github.com/Expensify/App/issues/37151 so that all collect policies created in NewDot are accessible in NewDot
- Should be coordinated to be released along with https://github.com/Expensify/App/issues/37151

This will make sure that any workspace created from NewDot is a collect workspace and not a free workspace.

- Change `CONST.POLICY.TYPE.FREE` to `CONST.POLICY.TYPE.TEAM` [here](https://github.com/Expensify/App/blob/2254c6688dc54850c5dff2b5e8f987093edb0dae/src/libs/actions/Policy.ts#L1245), [here](https://github.com/Expensify/App/blob/2254c6688dc54850c5dff2b5e8f987093edb0dae/src/libs/actions/Policy.ts#L1306), and [here](https://github.com/Expensify/App/blob/2254c6688dc54850c5dff2b5e8f987093edb0dae/src/libs/actions/Policy.ts#L1495)
- Update tests in [SidebarFilterTest](https://github.com/Expensify/App/blob/2254c6688dc54850c5dff2b5e8f987093edb0dae/tests/unit/SidebarFilterTest.js) and [PolicyTest](https://github.com/Expensify/App/blob/2254c6688dc54850c5dff2b5e8f987093edb0dae/tests/actions/PolicyTest.js#L55) to use `CONST.POLICY.TYPE.TEAM`
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016d89cd1132beada5</li>
        <li>Upwork Job ID: 1762127738310324224</li>
        <li>Last Price Increase: 2024-03-06</li>
<li>Automatic offers: </li>
<ul>
<li>GandalfGwaihir | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by slafortune** [comment]
@allroundexperts I see this is still on hold, but have you had a chance to review either proposal?

**Comment by allroundexperts** [comment]
@slafortune I think @tgolen has pretty much shared all the pieces that are needed in the issue description. @godofoutcasts94's [proposal](https://github.com/Expensify/App/issues/37153#issuecomment-1964418275) is essentially a duplicate of  what @codinggeek2023 [proposed](https://github.com/Expensify/App/issues/37153#issuecomment-1961815600) earlier. But even that proposal is just a duplicate of what @tgolen said [here](https://github.com/Expensify/App/issues/37153#issue-2151455345). I think we need a better proposal that has at-least a working demo.

**Comment by godofoutcasts94** [comment]
> @slafortune I think @tgolen has pretty much shared all the pieces that are needed in the issue description. @godofoutcasts94's [proposal](https://github.com/Expensify/App/issues/37153#issuecomment-1964418275) is essentially a duplicate of what @codinggeek2023 [proposed](https://github.com/Expensify/App/issues/37153#issuecomment-1961815600) earlier. But even that proposal is just a duplicate of what @tgolen said [here](https://github.com/Expensify/App/issues/37153#issue-2151455345). I think we need a better proposal that has at-least a working demo.

Should I attach the working video?

**Comment by slafortune** [comment]
Still on hold

**Comment by tgolen** [comment]
This is off HOLD

**Comment by slafortune** [comment]
@allroundexperts when you get a chance can you review the updated proposal?

**Comment by allroundexperts** [comment]
Will review this today.

**Comment by allroundexperts** [comment]
Reviewing again today.

**Comment by allroundexperts** [comment]
@GandalfGwaihir's [proposal](https://github.com/Expensify/App/issues/37153#issuecomment-1981200381) works well!

🎀 👀 🎀 C+ reviewed

**Comment by slafortune** [comment]
Looks like we are waiting on 38191 to be merged 👍

**Comment by slafortune** [comment]
@GandalfGwaihir  - Paid ✅ 

@allroundexperts can you please complete the checklist?

**Comment by allroundexperts** [comment]
@slafortune This was a new feature and as such, a checklist is not needed here. We can add the following regression test though:

1. Open the app and click on the `+` icon in the LHN.
2. Select `Create new workspace`. 
3. Open the workspaces page from settings.
4. Verify that the created workspace is of type collect.

Do we 👍 or 👎 ?

**Comment by slafortune** [comment]
@allroundexperts requires payment through NewDot Manual Requests -$500

**Comment by JmillsExpensify** [comment]
$500 approved for @allroundexperts
